query-connector/src/app/queryBuilding/buildFromTemplates/page.tsx

```diff
@@ -75,13 +80,28 @@ export default function QueryTemplateSelection() {
     };
   }, [selectedConditions, queryName]);
 
-  function handleCreateQueryClick(event: React.MouseEvent<HTMLButtonElement>) {
-    event.preventDefault();
-
-    validateForm();
-    if (!!queryName && !formError.queryName && !formError.selectedConditions) {
-      submitForm();
-    }
+  // ensures the fetchedConditions' checkbox statuses match
+  // the data in selectedCondtiions
+  function updateFetchedConditionIncludeStatus(
+    selectedConditions: CategoryNameToConditionOptionMap,
+  ) {
+    const prevFetch = structuredClone(fetchedConditions);
+    Object.entries(selectedConditions).map(
+      ([category, conditionsByCategory]) => {
+        Object.entries(conditionsByCategory).flatMap(
+          ([conditionId, conditionObj]) => {
+            if (prevFetch) {
+              const prevValues = prevFetch[category][conditionId];
```

Comment:
Just to check my understanding, since the `selectedConditions` that we're passing in / we're deriving the values of `category` and `conditionId` from are a subset of `fetchedConditions`, we wouldn't be in danger of a null pointer issue here, would we?

Reply:
correct! and that's another reason I wanted to stick with the existing data structures - if users tab back and forth between pages, want to make sure we're keeping everything as close to the same shape as possible so we're not trying to apply changes made on one page in a different shape to the data as it existed on the previous page